common/src/com/thoughtworks/go/util/UrlUtil.java

```diff
@@ -119,7 +119,9 @@ public static String getQueryParamFromUrl(String url, String paramName) {
             String[] split = query.split(QUERY_SEPERATOR);
             for (String queryFrag : split) {
                 String[] queryFragmentSplit = queryFrag.split(QUERY_KEY_VAL_SEPERATOR);
-                parsed.add(new QueryTuple(queryFragmentSplit[0], decode(queryFragmentSplit[1])));
+                if (queryFragmentSplit.length == 2) {
+                    parsed.add(new QueryTuple(queryFragmentSplit[0], decode(queryFragmentSplit[1])));
+                }
```

Comment:
This will delete query param failing this test
```
assertThat(UrlUtil.urlWithQuery("http://baz.quux?bang&hello=world", "foo", "bar"), is("http://baz.quux/?bang&hello=world&foo=bar"));
```

It's valid to have query param without a value. 
A discussion about that http://stackoverflow.com/questions/4557387/is-a-url-query-parameter-valid-if-it-has-no-value

Here's how it's used

https://github.com/gocd/gocd/blob/53de3722953779500d73f6ed6102dab08f348b8c/server/webapp/WEB-INF/rails.new/app/controllers/admin_controller.rb#L42

I think we should modify the code to **not** drop the query param.
Also, do you have more of the stack trace so that we can figure out the url being parsed?